roomserver/internal/api.go

```diff
@@ -2,20 +2,28 @@ package internal
 
 import (
 	"context"
-	"sync"
 
 	"github.com/Shopify/sarama"
 	fsAPI "github.com/matrix-org/dendrite/federationsender/api"
 	"github.com/matrix-org/dendrite/internal/caching"
 	"github.com/matrix-org/dendrite/internal/config"
 	"github.com/matrix-org/dendrite/roomserver/api"
+	"github.com/matrix-org/dendrite/roomserver/internal/input"
 	"github.com/matrix-org/dendrite/roomserver/internal/perform"
+	"github.com/matrix-org/dendrite/roomserver/internal/query"
 	"github.com/matrix-org/dendrite/roomserver/storage"
 	"github.com/matrix-org/gomatrixserverlib"
 )
 
 // RoomserverInternalAPI is an implementation of api.RoomserverInternalAPI
 type RoomserverInternalAPI struct {
+	*input.Inputer
+	*query.Queryer
```

Comment:
The names sit a bit funny with me at the moment, not least because of `Inputer` and `Queryer`. Maybe we can just stick with `Inputs`, `Queries`, `Invites`, `Joins` etc.

Reply:
I want to interface them actually so the -er suffix makes more sense then, which would allow us to test parts of the RSAPI more easily. Happy to have the concrete impl be `Inputs` `Queries` etc.

Reply:
Merging for now, will resolve later down the line.